homeassistant/components/camera/media_source.py

```diff
@@ -30,14 +29,13 @@ async def async_get_media_source(hass: HomeAssistant) -> CameraMediaSource:
 def _media_source_for_camera(
     hass: HomeAssistant, camera: Camera, content_type: str
 ) -> BrowseMediaSource:
+    assert (camera_state := hass.states.get(camera.entity_id)) is not None
```

Comment:
This seems wrong... What is Python runs in optimized mode and the asserts are not ran?

Then we don't fetch this at all?

Reply:
PS: We can rewrite this with having a need to assert at all.

Avoid the use of cast and assert whenever possible.

Reply:
How would you do it? Raise if `None`?